src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -738,4 +735,148 @@ private String assignUserRole(Map<String, Object> requestBody, List<String> exis
+						Map<String, Object> personalDetails = new HashMap<String, Object>();
+						personalDetails.put(Constants.FIRSTNAME.toLowerCase(), requestData.get(Constants.FIRST_NAME));
+						personalDetails.put(Constants.SURNAME, requestData.get(Constants.LAST_NAME));
+						personalDetails.put(Constants.PRIMARY_EMAIL, userData.get(Constants.EMAIL));
+						personalDetails.put(Constants.USER_NAME, userData.get(Constants.USER_NAME));
+						profileDetails.put(Constants.PERSONAL_DETAILS, personalDetails);
+
+						Map<String, Object> professionDetailObj = new HashMap<String, Object>();
+						professionDetailObj.put(Constants.ORGANIZATION_TYPE, Constants.GOVERNMENT);
+						if (StringUtils.isNotEmpty((String) userData.get(Constants.POSITION))) {
+							professionDetailObj.put(Constants.DESIGNATION, userData.get(Constants.POSITION));
+						}
+						List<Map<String, Object>> professionalDetailsList = new ArrayList<Map<String, Object>>();
+						professionalDetailsList.add(professionDetailObj);
+						profileDetails.put(Constants.PROFESSIONAL_DETAILS, professionalDetailsList);
+
+						updateRequestBody.put(Constants.PROFILE_DETAILS, profileDetails);
+						updateRequest.put(Constants.REQUEST, updateRequestBody);
+						Map<String, Object> updateReadData = (Map<String, Object>) outboundRequestHandlerService
+								.fetchResultUsingPatch(props.getSbUrl() + props.getLmsUserUpdatePath(), updateRequest, getDefaultHeaders());
+
+						if (Constants.OK.equalsIgnoreCase((String) updateReadData.get(Constants.RESPONSE_CODE))) {
+							Map<String,Object> roleMap=new HashMap<>();
+							roleMap.put(Constants.ORGANIZATION_ID,userData.get(Constants.ROOT_ORG_ID));
+							roleMap.put(Constants.USER_ID,userId);
+							retValue=assignRole(roleMap);
+						}
+					}
+				}
+
+		} catch (Exception e) {
+			errMsg = "Failed to process message. Exception: " + e.getMessage();
+		}
+		if (StringUtils.isNotBlank(errMsg) || retValue==false) {
+			response.getParams().setStatus(Constants.FAILED);
+			response.getParams().setErrmsg(errMsg);
+			response.setResponseCode(HttpStatus.BAD_REQUEST);
+		}
+		return response;
+
+	}
+	
+	public Map<String, Object> getUsersReadData(String userId, String authToken, String userAuthToken) {
+		Map<String, String> header = new HashMap<>();
+		if (StringUtils.isNotEmpty(authToken)) {
+			header.put(Constants.AUTH_TOKEN, authToken);
+		}
+		if (StringUtils.isNotEmpty(userAuthToken)) {
+			header.put(Constants.X_AUTH_TOKEN, userAuthToken);
+		}
+		Map<String, Object> readData = (Map<String, Object>) outboundRequestHandlerService
+				.fetchUsingGetWithHeaders(serverConfig.getSbUrl() + serverConfig.getLmsUserReadPath() + userId,
+						header);
+
+		Map<String, Object> result = (Map<String, Object>) readData.get(Constants.RESULT);
+		Map<String, Object> responseMap = (Map<String, Object>) result.get(Constants.RESPONSE);
+		return responseMap;
+	}
+
+	public boolean assignRole(Map<String,Object> request) throws Exception{
```

Comment:
this method is not needed

Reply:
make it as private method inside ProfileService